### PR TITLE
Deeper merge

### PR DESCRIPTION
**Pull request by Sodman**

### CODE REVIEW

File: pkg/api/v2/reporter/reporter.go [hunk 1]
```diff
@@ -30,9 +30,63 @@ func (e ResourceReports) Accept(res ...resources.InputResource) ResourceReports
...
+				}
+			}
+
+			if firstReport.Errors == nil {
+				// Only 2nd has errs
+				firstReport.Errors = secondReport.Errors
+				e[k] = firstReport
+				continue
+			} else if secondReport.Errors == nil {
+				// Only 1st has errs
```
Comment (kdorosh) [inline pkg/api/v2/reporter/reporter.go:62]: don't we still need `e[k] = firstReport` here, in the event we had warnings but no errors in the second report?

File: pkg/api/v2/reporter/reporter.go [hunk 2]
```diff
@@ -30,9 +30,63 @@ func (e ResourceReports) Accept(res ...resources.InputResource) ResourceReports
...
+						allErrsMap[err.Error()] = err
+					}
+					for _, err := range errs2.Errors {
+						if _, found := allErrsMap[err.Error()]; !found {
+							allErrsMap[err.Error()] = err
+							errs1.Errors = append(errs1.Errors, err)
+						}
+					}
+				}
+				firstReport.Errors = errs1
```
Comment (kdorosh) [inline pkg/api/v2/reporter/reporter.go:82]: what if `errs1` is multi but `errs2` is not?

Comment (kdorosh) [reply pkg/api/v2/reporter/reporter.go:82]: and vice-versa

File: pkg/api/v2/reporter/reporter.go (hunk 1 above)
Comment (Sodman) [reply pkg/api/v2/reporter/reporter.go:62]: Great catch, added this + added a test for this case!

File: pkg/api/v2/reporter/reporter.go (hunk 2 above)
Comment (Sodman) [reply pkg/api/v2/reporter/reporter.go:82]: As far as I'm aware these should always be multierrs or nil... but if either (or both) of them is a non-nil non-multi err, what would be the expected behavior here? Concat the error strings?

Comment (Sodman) [reply pkg/api/v2/reporter/reporter.go:82]: I guess if one is multi and the other is not, we can add the non-multi to the multi... and if both are just regular errors then we concat?

Comment (kdorosh) [reply pkg/api/v2/reporter/reporter.go:82]: if they both aren't multis, we could just make a multi

File: pkg/api/v2/reporter/reporter.go [hunk 3]
```diff
@@ -30,9 +30,71 @@ func (e ResourceReports) Accept(res ...resources.InputResource) ResourceReports
...
+			errs1, isFirstMulti := firstReport.Errors.(*multierror.Error)
+			errs2, isSecondMulti := secondReport.Errors.(*multierror.Error)
+
+			// If the errors are not mutliErrs, wrap them in multiErrs:
+			if !isFirstMulti {
+				errs1 = &multierror.Error{Errors: []error{firstReport.Errors}}
+			}
+			if !isSecondMulti {
+				errs2 = &multierror.Error{Errors: []error{secondReport.Errors}}
+			}
```
Comment (kdorosh) [inline pkg/api/v2/reporter/reporter.go:79]: we shouldn't need to do this, `multierror.Append` handles this for us already

**Review by kdorosh — APPROVED** [review]

**Pull request merged**